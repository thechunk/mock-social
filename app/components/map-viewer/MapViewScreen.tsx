import * as React from "react";
import MapView from "./MapView";
import {PureComponent} from "react";
import {NavigationInjectedProps} from "react-navigation";
import withNavigationParamsAsProps from "../../hoc/withNavigationParamsAsProps";
import styles from "./styles";

export class MapViewScreen extends PureComponent<IMapViewScreen & NavigationInjectedProps> {
    render() {
        const {lat, lng} = this.props.hydrate.address.geo;
        return <MapView
            region={{
                latitude: lat,
                longitude: lng,
                latitudeDelta: 0.8,
                longitudeDelta: 0.8
            }}
            style={styles.view} />;
    }
}

export default withNavigationParamsAsProps(MapViewScreen);
