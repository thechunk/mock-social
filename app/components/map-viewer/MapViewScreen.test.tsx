import * as React from 'react';
import {shallow} from "enzyme";
import {MapViewScreen} from "./MapViewScreen";

describe('MapViewScreen', () => {
    it('renders correctly', () => {
        expect(shallow(<MapViewScreen hydrate={{address: {geo: {lat: 1.2, lng: 2.3}}}} />)).toMatchSnapshot();
    });
});