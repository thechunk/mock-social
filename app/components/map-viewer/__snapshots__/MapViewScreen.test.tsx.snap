// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapViewScreen renders correctly 1`] = `
<_class
  region={
    Object {
      "latitude": 1.2,
      "latitudeDelta": 0.8,
      "longitude": 2.3,
      "longitudeDelta": 0.8,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
